server/tournaments/index.js

```diff
@@ -781,7 +781,12 @@ class Tournament {
 
 		// Prevent battles between offline users from starting
 		const from = Users.get(challenge.from.userid);
-		if (!from || !from.connected || !user.connected) return;
+		if (!from || !user.connected) return;
+		if (!from.connected) {
+			// Prevent an exploit
+			this.cancelChallenge(from);
```

Comment:
Seems like this should be handled in ``User#onDisconnect()``

Reply:
`User` objects don't store any reference to room games they're in, so we'd have to iterate through every single room on the simulator looking for tours the user is in and cancel their challenges. After all, there's nothing stopping a user from leaving a room while they're entered in its tournament.

Reply:
It's ``user.games``

Reply:
Also like you should just cancel all challenges on disconnect; roomgames in theory shouldn't come into it.